### ISSUE TITLE
doesn't handle locales without regions very well 

### ISSUE THREAD

**Issue by thedude61636** [long1eu/flutter_i18n#34]
currently i'm using the library with only the language code without the country code
library generates the i18n file as follows 

![image](https://user-images.githubusercontent.com/13031288/50541431-0a39ed80-0bb7-11e9-89db-df8e180c9539.png)

currently flutter for some reason when using a locale with a language code and an empty country code results in creating a locale languageCode_countryCode
so when creating Locale("ar,"")  it gives ar_ 
and when using that locale with the material localizations it's says it's not supported 
and that crashes the app when using it on ios specifically when long pressing a textfield to show the interactive selection menu as it can't translate the texts 
tl;dr when creating the Locale in the supported locales it should be Locale("ar",null) 
if the locale doesn't have a country

**Comment by thedude61636** [comment]
after 3 hours of searching it turns out it's flutter's fault 
https://github.com/flutter/flutter/issues/13452

never the less if you have the time please do remove the empty country code
